webshop-backend/src/shopping/ShoppingCart.java

```diff
@@ -23,8 +23,11 @@ public User getOwner() {
     public boolean addProduct(Product p) {
         boolean ok = false;
         if (p != null) {
-            products.add(p);
-            p.decreaseStock();
+            //nem lehet 0 a stock, mert abból nem tudok levonni darabszámot
+            if(p.getStock()!=0){
+                products.add(p);
+                p.decreaseStock();
+            }
             ok = true;
```

Comment:
why is `ok = true;` outside the `if` statement ?
I'm sure you don't want to return `true` if the `if(p.getStock()!=0)` statement is false.
Or am I wrong ?

Reply:
Right. I will update shoppingCart if it is allowed in the validator branch?! 
Did you read my review?
I would like to write a setProductListEmpty(){products.clear()} method.

Reply:
@deb05121, sure, you can fix these one-liners here. If you notice something that requires more job, it is better to open a new branch, but in the case of these 1-line changes, I think it is okay :)